src/main/java/com/aws/greengrass/clientdevices/auth/certificate/infra/BackgroundCertificateRefresh.java

```diff
@@ -208,7 +264,12 @@ private void refreshCertificateValidity(Certificate certificate) {
         }
 
         VerifyIotCertificate useCase = useCases.get(VerifyIotCertificate.class);
-        useCase.apply(certPem.get());
+        try {
+            useCase.apply(certPem.get());
```

Comment:
When is this throwing a RuntimeException? We should fix that

Reply:
Here is the reasoning. Currently it does not if you check the code but if someone were to make changes that introduce that and we don't catch it here we run the risk of the thread dying and not getting rescheduled for a next run